web/handler.go

```diff
@@ -0,0 +1,217 @@
+package web
+
+import (
+	"fmt"
+	"html"
+	"net/http"
+	"net/http/pprof"
+	"sync"
+
+	"github.com/criteo/graphite-remote-adapter/client"
+	"github.com/criteo/graphite-remote-adapter/client/graphite"
+	"github.com/criteo/graphite-remote-adapter/config"
+	"github.com/criteo/graphite-remote-adapter/ui"
+	"github.com/criteo/graphite-remote-adapter/utils/template"
+	"github.com/davecgh/go-spew/spew"
+	assetfs "github.com/elazarl/go-bindata-assetfs"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/version"
+)
+
+const namespace = "remote_adapter"
+
+var (
+	requestCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "api_requests_total",
+			Help: "A counter for requests to the wrapped handler.",
+		},
+		[]string{"handler", "code", "method"},
+	)
+	requestDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "request_duration_seconds",
+			Help:    "A histogram of latencies for requests.",
+			Buckets: []float64{.25, .5, 1, 2.5, 5, 10},
+		},
+		[]string{"handler", "method"},
+	)
+	responseSize = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "response_size_bytes",
+			Help:    "A histogram of response sizes for requests.",
+			Buckets: []float64{200, 500, 900, 1500},
+		},
+		[]string{"handler"},
+	)
+)
+
+// Handler serves various HTTP endpoints of the remote adapter server
+type Handler struct {
+	logger log.Logger
+
+	cfg      *config.Config
+	router   *mux.Router
+	reloadCh chan chan error
+
+	writers []client.Writer
+	readers []client.Reader
+
+	lock sync.RWMutex
+}
+
+func instrumentHandler(name string, handlerFunc http.HandlerFunc) http.Handler {
+	return promhttp.InstrumentHandlerDuration(
```

Comment:
wasn't there something do do this in the prometheus lib ?

Reply:
It has been deprecated and explained here: https://github.com/prometheus/client_golang/blob/master/prometheus/http.go#L168

Reply:
ah ok !